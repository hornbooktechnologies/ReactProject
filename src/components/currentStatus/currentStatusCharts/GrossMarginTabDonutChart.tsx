import CustomPie from "../../common/customPie/CustomPie";

function PlanSeries(series: any) {
    return (
    <div className="uk-flex uk-width-1-1 mb-20">
            <label className="uk-width-1-2 uk-text-right pr-24">
              {series.title}
            </label>
            <p
              className="uk-margin-remove-top uk-width-1-2 uk-text-left mt-5 mb-10 font-14"
            >
              <strong>
                {series.prefix ? <>{series.prefix}</> : ""}
                <>
                  {series.seriesValue !== "" ? (
                    <>{series.seriesValue.toLocaleString("en")}</>
                  ) : (
                    ""
                  )}
                  {series.suffix1 ? (
                    <span className="uk-text-top font-10">
                      {series.suffix1}
                    </span>
                  ) : (
                    ""
                  )}
                </>
              </strong>
            </p>
          </div>
    )
}

function ForecastSeries(series: any) {
    return (
        <div className="uk-flex uk-width-1-1 mb-20">
            <label className="uk-width-1-2 uk-text-right pr-24">
              {series.title}
            </label>
            <p
              className=
                "uk-margin-remove-top uk-width-1-2 uk-text-left mt-5 mb-10 font-14"
            >
              <strong>
                {series.prefix ? <>{series.prefix}</> : ""}
                <>
                  {series.seriesValue !== "" ? (
                    <>{series.seriesValue.toLocaleString("en")}</>
                  ) : (
                    ""
                  )}
                  {series.suffix1 ? (
                    <span className="uk-text-top font-10">
                      {series.suffix1}
                    </span>
                  ) : (
                    ""
                  )}
                </>
              </strong>
            </p>
          </div>
    )
}

function ActualSeries(actualSeries: any) {
    return (
        <div className="uk-flex uk-width-1-1 mb-20">
          <label className="uk-width-1-2 uk-text-right pr-24">Actual</label>
          <p className="uk-margin-remove-top uk-margin-remove-bottom uk-width-1-2 uk-text-left">
            <strong>
              {actualSeries.Prefix ? <>{actualSeries.Prefix}</> : ""}
              <>
                {actualSeries.Actual !== "" ? (
                  <>{actualSeries.Actual?.toLocaleString("en")}</>
                ) : (
                  ""
                )}
                {actualSeries.Suffix ? (
                  <span className="uk-text-top font-10">{actualSeries.Suffix}</span>
                ) : (
                  ""
                )}
              </>
            </strong>
          </p>
        </div>
      );
}

const GrossMarginTabDonutChart = (props: any) => {
    return (
        <>
        <div className="uk-flex uk-flex-middle uk-height-1-1">
            <div className="uk-width-1-1">
                <PlanSeries {...props.data.series1} />
                <ForecastSeries {...props.data.series2} />
                <ActualSeries { ...props.actualData } />
            </div>
            <div className="uk-flex uk-flex-center">
                <CustomPie {...props} />
            </div>
        </div>
        </>
    )
}

export default GrossMarginTabDonutChart;