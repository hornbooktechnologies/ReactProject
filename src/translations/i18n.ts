import { initReactI18next } from 'react-i18next'
import i18n from 'i18next'
import LanguageDetector from 'i18next-browser-languagedetector'
import { TRANSLATIONS_EN } from './en/Translations'
import { TRANSLATIONS_JA } from './ja/Translations'

i18n
  .use(LanguageDetector)
  .use(initReactI18next)
  .init({
    lng: 'en',
    resources: {
      en: {
        translation: TRANSLATIONS_EN,
      },
      ja: {
        translation: TRANSLATIONS_JA,
      },
    },
  })

export default i18n
