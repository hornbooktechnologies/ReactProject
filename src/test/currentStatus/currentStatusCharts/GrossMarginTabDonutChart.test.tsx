import React from "react";
import GrossMarginTabDonutChart from "../../../components/currentStatus/currentStatusCharts/GrossMarginTabDonutChart";
import renderer from "react-test-renderer";
import i18n from '../../i18nForTests'
import { I18nextProvider } from 'react-i18next'

describe("Component: GrossMarginTabDonutChart", () => {
  const props = {
    actualData: {
      Actual: 304,
      Forecast: 970,
      Plan: 738,
      Prefix: "¥",
      Suffix: "Oku",
      Suffix2: "Oku",
    },
    data: {
      chartGradientType: "green",
      color: "#66A428",
      forecast: 970,
      gradientColor1: "#9FCD72",
      gradientColor2: "#66A428",
      header: "Gross Margin",
      plan: 738,
      series1: {
        prefix: "¥",
        seriesValue: 738,
        suffix1: "Oku",
        suffix2: "Oku",
        title: "Plan",
      },
      series2: {
        prefix: "¥",
        seriesValue: 970,
        suffix1: "Oku",
        suffix2: "Oku",
        title: "Forecast",
      },
      type: "large",
      value: 131.43631436314362,
    },
    height: 150,
    width: 150,
  };
  it("GrossMarginTabDonutChart renders correctly", () => {
    const tree = renderer
      .create(
        <I18nextProvider i18n={i18n}>
          <GrossMarginTabDonutChart {...props} />
        </I18nextProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
