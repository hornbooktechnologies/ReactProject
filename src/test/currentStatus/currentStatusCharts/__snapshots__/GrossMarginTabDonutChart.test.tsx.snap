// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: GrossMarginTabDonutChart GrossMarginTabDonutChart renders correctly 1`] = `
<div
  className="uk-flex uk-flex-middle uk-height-1-1"
>
  <div
    className="uk-width-1-1"
  >
    <div
      className="uk-flex uk-width-1-1 mb-20"
    >
      <label
        className="uk-width-1-2 uk-text-right pr-24"
      >
        Plan
      </label>
      <p
        className="uk-margin-remove-top uk-width-1-2 uk-text-left mt-5 mb-10 font-14"
      >
        <strong>
          ¥
          738
          <span
            className="uk-text-top font-10"
          >
            Oku
          </span>
        </strong>
      </p>
    </div>
    <div
      className="uk-flex uk-width-1-1 mb-20"
    >
      <label
        className="uk-width-1-2 uk-text-right pr-24"
      >
        Forecast
      </label>
      <p
        className="uk-margin-remove-top uk-width-1-2 uk-text-left mt-5 mb-10 font-14"
      >
        <strong>
          ¥
          970
          <span
            className="uk-text-top font-10"
          >
            Oku
          </span>
        </strong>
      </p>
    </div>
    <div
      className="uk-flex uk-width-1-1 mb-20"
    >
      <label
        className="uk-width-1-2 uk-text-right pr-24"
      >
        Actual
      </label>
      <p
        className="uk-margin-remove-top uk-margin-remove-bottom uk-width-1-2 uk-text-left"
      >
        <strong>
          ¥
          304
          <span
            className="uk-text-top font-10"
          >
            Oku
          </span>
        </strong>
      </p>
    </div>
  </div>
  <div
    className="uk-flex uk-flex-center"
  >
    <div
      className="uk-flex uk-flex-center"
    >
      <div
        className="chart-continer"
        style={
          Object {
            "height": 150,
            "width": 150,
          }
        }
      >
        <div
          className="dount-chart-container"
        >
          <div />
        </div>
        <div
          className=" donutmiddleData"
        >
          <div
            className="uk-text-center thumb-font"
          >
            <div
              className="jera-like-font"
            >
              <i
                className="jera color-green-60 jera-like-1"
              />
            </div>
            <div
              className="font-prvalue color-green-60"
            >
              <span>
                131.44
              </span>
              <span
                className="font-12 vertical-top"
              >
                %
              </span>
            </div>
            <div>
              <span
                className="font-12 totalRatio"
              >
                CHARTS.DONUT.TOOLTIP_HEAD
              </span>
              <span
                className="uk-tooltip-cutom tooltip-text"
              >
                <i
                  className="jera-info-circle icn-16 color-blue-70 top-2 ml-3"
                />
                <div
                  className="uk-tooltip uk-animation-scale-up uk-animation-center uk-transdorm-origin-center-left uk-togglabe-enter uk-tooltip-bottom uk-tooltip-bottom-center uk-transform-origin-center-left uk-text-left font-10"
                >
                  <span
                    className="color-grey-blue-80 uk-display-block"
                  >
                    <strong
                      className="color-grey-blue-95"
                    >
                      CHARTS.DONUT.TOOLTIP_MAIN_TEXT
                    </strong>
                      
                    CHARTS.DONUT.TOOLTIP_SUB_TEXT
                  </span>
                  <span
                    className="color-red-60 mt-15 uk-display-block"
                  >
                    CHARTS.DONUT.EQUATION
                  </span>
                  <span
                    className="color-grey-blue-80 uk-display-block"
                  >
                    CHARTS.DONUT.EQUATION_LHS
                     = Y (
                    Forecast
                    ) / X (
                    Plan
                    )
                  </span>
                </div>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
